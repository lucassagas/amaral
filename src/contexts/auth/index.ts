export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
