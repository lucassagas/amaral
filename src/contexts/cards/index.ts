export { CardContext, CardContextProvider } from './cards';

