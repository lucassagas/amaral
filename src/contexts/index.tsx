import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
