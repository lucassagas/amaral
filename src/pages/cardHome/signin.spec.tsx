import { AppProvider } from '@/contexts/index'
import { act, fireEvent, render, screen } from '@testing-library/react'
import { GetServerSidePropsContext } from 'next'
import SignIn, { getServerSideProps } from './index'

describe('<Signin />', () => {
  it('should render signin page correctly', () => {
    const { container } = render(
      <AppProvider>
        <SignIn />
      </AppProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('should call signin function', async () => {
    render(
      <AppProvider>
        <SignIn />
      </AppProvider>
    )

    const email = 'lucassaagas@gmail.com'

    const emailInput: HTMLInputElement = screen.getByRole('textbox', {
      name: /e-mail senha/i,
    })

    const button = screen.getByRole('button', {
      name: /entrar/i,
    })

    act(() => {
      fireEvent.change(emailInput, { target: { value: email } })
    })

    await act(async () => {
      fireEvent.click(button)
    })

    expect(emailInput.value).toEqual(email)
  })

  it('should test server side props ', async () => {
    const response = await getServerSideProps({} as GetServerSidePropsContext)

    expect(response).toEqual(
      expect.objectContaining({
        props: {},
      })
    )
  })
})
