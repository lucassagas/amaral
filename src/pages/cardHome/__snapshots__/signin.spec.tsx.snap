// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin /> should render signin page correctly 1`] = `
<div>
  <div
    class="css-1xjsnt2"
  >
    <form
      class="css-1xyw26x"
    >
      <img
        alt="logo"
        class="chakra-image css-gb7ck4"
        src="logo.png"
      />
      <div
        class="chakra-stack css-egoftb"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-pk5tju"
            for="email"
            id="field-1-label"
          >
            E-mail
          </label>
          <input
            class="chakra-input css-uxk25m"
            id="email"
            name="email"
            type="email"
          />
        </div>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-pk5tju"
            for="email"
            id="field-2-label"
          >
            Senha
          </label>
          <input
            class="chakra-input css-uxk25m"
            id="password"
            name="password"
            type="password"
          />
        </div>
      </div>
      <button
        class="chakra-button css-1y4zvm6"
        type="submit"
      >
        Entrar
      </button>
    </form>
  </div>
</div>
`;
